
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Functions for postprocessing lists of mutations &#8212; isomut2py 2.0.1 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="_static/icon.ico"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Technical (formatting, IO, etc.) functions" href="code_technical_functions.html" />
    <link rel="prev" title="Functions for plotting" href="code_plotting_functions.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="module-isomut2py.postprocess">
<span id="functions-for-postprocessing-lists-of-mutations"></span><h1>Functions for postprocessing lists of mutations<a class="headerlink" href="#module-isomut2py.postprocess" title="Permalink to this headline">¶</a></h1>
<dl class="function">
<dt id="isomut2py.postprocess.calculate_DNV_matrix">
<code class="descclassname">isomut2py.postprocess.</code><code class="descname">calculate_DNV_matrix</code><span class="sig-paren">(</span><em>mutations_dataframe=None</em>, <em>sample_names=None</em>, <em>unique_only=True</em>, <em>output_dir=None</em>, <em>mutations_filename=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.postprocess.calculate_DNV_matrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the DNV matrix from a dataframe of relevant mutations.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>mutations_dataframe</strong> – Dataframe containing the list of mutations to be considered. (default: None) (pandas.DataFrame)</li>
<li><strong>sample_names</strong> – the list of sample names included in the analysis (default: None) (list of str)</li>
<li><strong>unique_only</strong> – if True, only unique mutations are considered for the spectrum (default: True) (bool)</li>
<li><strong>output_dir</strong> – path to the directory where mutation tables are located (default: None) (str)</li>
<li><strong>mutations_filename</strong> – path to the mutation table(s) (default: None) (list of str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">A dictionary containing DNV matrices as values and sample names as keys. (str: numpy.array)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.postprocess.calculate_DNV_spectrum">
<code class="descclassname">isomut2py.postprocess.</code><code class="descname">calculate_DNV_spectrum</code><span class="sig-paren">(</span><em>mutations_dataframe=None</em>, <em>sample_names=None</em>, <em>unique_only=True</em>, <em>output_dir=None</em>, <em>mutations_filename=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.postprocess.calculate_DNV_spectrum" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the indel spectrum from a dataframe of relevant mutations.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>mutations_dataframe</strong> – Dataframe containing the list of mutations to be considered. (default: None) (pandas.DataFrame)</li>
<li><strong>sample_names</strong> – the list of sample names included in the analysis (default: None) (list of str)</li>
<li><strong>unique_only</strong> – if True, only unique mutations are considered for the spectrum (default: True) (bool)</li>
<li><strong>output_dir</strong> – path to the directory where mutation tables are located (default: None) (str)</li>
<li><strong>mutations_filename</strong> – path to the mutation table(s) (default: None) (list of str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">A dictionary containing DNV spectra arrays (the counts for each mutation type) as values and sample names as keys. (str: numpy.array)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.postprocess.calculate_SNV_spectrum">
<code class="descclassname">isomut2py.postprocess.</code><code class="descname">calculate_SNV_spectrum</code><span class="sig-paren">(</span><em>ref_fasta</em>, <em>mutations_dataframe=None</em>, <em>sample_names=None</em>, <em>unique_only=True</em>, <em>chromosomes=None</em>, <em>output_dir=None</em>, <em>mutations_filename=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.postprocess.calculate_SNV_spectrum" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the triplet spectrum from a dataframe of relevant mutations, using the fasta file of the reference genome.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>ref_fasta</strong> – the path to the reference genome fasta file (str)</li>
<li><strong>mutations_dataframe</strong> – Dataframe containing the list of mutations to be considered. (default: None) (pandas.DataFrame)</li>
<li><strong>sample_names</strong> – the list of sample names included in the analysis (default: None) (list of str)</li>
<li><strong>unique_only</strong> – if True, only unique mutations are considered for the spectrum (default: True) (bool)</li>
<li><strong>chromosomes</strong> – list of chromosomes in the analysis (default: None) (list of str)</li>
<li><strong>output_dir</strong> – path to the directory where mutation tables are located (default: None) (str)</li>
<li><strong>mutations_filename</strong> – path to the mutation table(s) (default: None) (list of str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">A dictionary containing 96-element vectors (the counts for each mutation type) as values and sample names as keys. (str: numpy.array)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.postprocess.calculate_indel_spectrum">
<code class="descclassname">isomut2py.postprocess.</code><code class="descname">calculate_indel_spectrum</code><span class="sig-paren">(</span><em>ref_fasta</em>, <em>mutations_dataframe=None</em>, <em>sample_names=None</em>, <em>unique_only=True</em>, <em>chromosomes=None</em>, <em>output_dir=None</em>, <em>mutations_filename=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.postprocess.calculate_indel_spectrum" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the indel spectrum from a dataframe of relevant mutations, using the fasta file of the reference genome.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>ref_fasta</strong> – the path to the reference genome fasta file (str)</li>
<li><strong>mutations_dataframe</strong> – Dataframe containing the list of mutations to be considered. (default: None) (pandas.DataFrame)</li>
<li><strong>sample_names</strong> – the list of sample names included in the analysis (default: None) (list of str)</li>
<li><strong>unique_only</strong> – if True, only unique mutations are considered for the spectrum (default: True) (bool)</li>
<li><strong>chromosomes</strong> – list of chromosomes in the analysis (default: None) (list of str)</li>
<li><strong>output_dir</strong> – path to the directory where mutation tables are located (default: None) (str)</li>
<li><strong>mutations_filename</strong> – path to the mutation table(s) (default: None) (list of str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">A dictionary containing 83-element vectors (the counts for each mutation type) as values and sample names as keys. (str: numpy.array)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.postprocess.check_pileup">
<code class="descclassname">isomut2py.postprocess.</code><code class="descname">check_pileup</code><span class="sig-paren">(</span><em>chrom_list</em>, <em>from_pos_list</em>, <em>to_pos_list</em>, <em>ref_fasta</em>, <em>input_dir</em>, <em>bam_filename</em>, <em>output_dir=''</em>, <em>samtools_fullpath='samtools'</em>, <em>base_quality_limit=30</em>, <em>samtools_flags=' -B -d 1000 '</em>, <em>print_original=True</em>, <em>filename=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.postprocess.check_pileup" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads pileup information for a list of genomic regions.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>chrom_list</strong> – List of chromosomes for the regions. (list of str)</li>
<li><strong>from_pos_list</strong> – List of starting positions for the regions. (list of int)</li>
<li><strong>to_pos_list</strong> – List of ending positions for the regions. (list of int)</li>
<li><strong>ref_fasta</strong> – the path to the reference genome fasta file (str)</li>
<li><strong>bam_filename</strong> – the filenames for the bam files of the investigated samples (list of str)</li>
<li><strong>output_dir</strong> – path to the directory where temporary files should be saved (default: ‘’) (str)</li>
<li><strong>samtools_fullpath</strong> – path to samtools on the computer (default: “samtools’) (str)</li>
<li><strong>base_quality_limit</strong> – base quality limit for the pileup generation (default: 30) (int)</li>
<li><strong>samtools_flags</strong> – additional flags for samtools (default: ‘ -B -d ‘ + str(SAMTOOLS_MAX_DEPTH) + ‘ ‘) (str)</li>
<li><strong>print_original</strong> – If True, prints the original string generated with samtools mpileup as well (default: True) (bool)</li>
<li><strong>filename</strong> – If filename is specified, results will be writted to that path. Otherwise, results are written to [output_dir]/checkPileup_tmp.csv (default: None) (str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">df: The processed results of the pileup, containing information on all samples in a pandas.DataFrame.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.postprocess.decompose_DNV_spectra">
<code class="descclassname">isomut2py.postprocess.</code><code class="descname">decompose_DNV_spectra</code><span class="sig-paren">(</span><em>DNVspectrumDict</em>, <em>sample_names=None</em>, <em>signatures_file=None</em>, <em>equal_initial_proportions=False</em>, <em>tol=0.0001</em>, <em>max_iter=1000</em>, <em>filter_percent=0</em>, <em>filter_count=0</em>, <em>keep_top_n=None</em>, <em>use_signatures=None</em>, <em>ignore_signatures=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.postprocess.decompose_DNV_spectra" title="Permalink to this definition">¶</a></dt>
<dd><p>Run the whole pipeline of decomposing DNV spectra for the samples specified in sample_names.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>DNVspectrumDict</strong> – dictionary containing DNV spectra as values and sample names as keys (dictionary)</li>
<li><strong>sample_names</strong> – The list of sample names analysed. (list of str)</li>
<li><strong>unique_only</strong> – If True, only unique mutations are used to construct the original spectrum.</li>
<li><strong>signatures_file</strong> – The path to the csv file containing the signature matrix. (str)</li>
<li><strong>equal_initial_proportions</strong> – If True, initial weights are initialized to be equal for all reference signature. Otherwise, they are initialized so the signatures with large cosine similarity with the original spectrum get larger weights. (default: False) (bool)</li>
<li><strong>tol</strong> – The maximal difference between the proportions for convergence to be True. (default: 0.0001) (float)</li>
<li><strong>max_iter</strong> – The maximum number of iterations (default: 1000) (int)</li>
<li><strong>filter_percent</strong> – Filter signatures, that contribute less than filter_percent of the mutations in the sample. (default: 0) (float)</li>
<li><strong>filter_count</strong> – Filter signatures, that contribute less than filter_count number of mutations in the sample. (default: 0) (int)</li>
<li><strong>keep_top_n</strong> – Only keep those signatures that contribute to the mixture with the top keep_top_n number of mutation. (default: None) (int)</li>
<li><strong>use_signatures</strong> – Use a specific subset of all signatures. A list of signature names. (default: None) (list of str)</li>
<li><strong>ignore_signatures</strong> – Exclude a specific subset of all signatures from the analysis. A list of signature names. (default: None) (list of str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">The final proportions of all signatures in the mixture. (Filtered out or not used signatures appear with a proportion of zero.) (numpy.array)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.postprocess.decompose_SNV_spectra">
<code class="descclassname">isomut2py.postprocess.</code><code class="descname">decompose_SNV_spectra</code><span class="sig-paren">(</span><em>SNVspectrumDict</em>, <em>sample_names=None</em>, <em>signatures_file=None</em>, <em>equal_initial_proportions=False</em>, <em>tol=0.0001</em>, <em>max_iter=1000</em>, <em>filter_percent=0</em>, <em>filter_count=0</em>, <em>keep_top_n=None</em>, <em>use_signatures=None</em>, <em>ignore_signatures=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.postprocess.decompose_SNV_spectra" title="Permalink to this definition">¶</a></dt>
<dd><p>Run the whole pipeline of decomposing SNV spectra for the samples specified in sample_names.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>SNVspectrumDict</strong> – dictionary containing SNV spectra as values and sample names as keys (dictionary)</li>
<li><strong>sample_names</strong> – The list of sample names analysed. (list of str)</li>
<li><strong>unique_only</strong> – If True, only unique mutations are used to construct the original spectrum.</li>
<li><strong>signatures_file</strong> – The path to the csv file containing the signature matrix. (str)</li>
<li><strong>equal_initial_proportions</strong> – If True, initial weights are initialized to be equal for all reference signature. Otherwise, they are initialized so the signatures with large cosine similarity with the original spectrum get larger weights. (default: False) (bool)</li>
<li><strong>tol</strong> – The maximal difference between the proportions for convergence to be True. (default: 0.0001) (float)</li>
<li><strong>max_iter</strong> – The maximum number of iterations (default: 1000) (int)</li>
<li><strong>filter_percent</strong> – Filter signatures, that contribute less than filter_percent of the mutations in the sample. (default: 0) (float)</li>
<li><strong>filter_count</strong> – Filter signatures, that contribute less than filter_count number of mutations in the sample. (default: 0) (int)</li>
<li><strong>keep_top_n</strong> – Only keep those signatures that contribute to the mixture with the top keep_top_n number of mutation. (default: None) (int)</li>
<li><strong>use_signatures</strong> – Use a specific subset of all signatures. A list of signature names. (default: None) (list of str)</li>
<li><strong>ignore_signatures</strong> – Exclude a specific subset of all signatures from the analysis. A list of signature names. (default: None) (list of str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">The final proportions of all signatures in the mixture. (Filtered out or not used signatures appear with a proportion of zero.) (numpy.array)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.postprocess.decompose_indel_spectra">
<code class="descclassname">isomut2py.postprocess.</code><code class="descname">decompose_indel_spectra</code><span class="sig-paren">(</span><em>IDspectrumDict</em>, <em>sample_names=None</em>, <em>signatures_file=None</em>, <em>equal_initial_proportions=False</em>, <em>tol=0.0001</em>, <em>max_iter=1000</em>, <em>filter_percent=0</em>, <em>filter_count=0</em>, <em>keep_top_n=None</em>, <em>use_signatures=None</em>, <em>ignore_signatures=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.postprocess.decompose_indel_spectra" title="Permalink to this definition">¶</a></dt>
<dd><p>Run the whole pipeline of decomposing indel spectra for the samples specified in sample_names.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>IDspectrumDict</strong> – dictionary containing indel spectra as values and sample names as keys (dictionary)</li>
<li><strong>sample_names</strong> – The list of sample names analysed. (list of str)</li>
<li><strong>signatures_file</strong> – The path to the csv file containing the signature matrix. (str)</li>
<li><strong>equal_initial_proportions</strong> – If True, initial weights are initialized to be equal for all reference signature. Otherwise, they are initialized so the signatures with large cosine similarity with the original spectrum get larger weights. (default: False) (bool)</li>
<li><strong>tol</strong> – The maximal difference between the proportions for convergence to be True. (default: 0.0001) (float)</li>
<li><strong>max_iter</strong> – The maximum number of iterations (default: 1000) (int)</li>
<li><strong>filter_percent</strong> – Filter signatures, that contribute less than filter_percent of the mutations in the sample. (default: 0) (float)</li>
<li><strong>filter_count</strong> – Filter signatures, that contribute less than filter_count number of mutations in the sample. (default: 0) (int)</li>
<li><strong>keep_top_n</strong> – Only keep those signatures that contribute to the mixture with the top keep_top_n number of mutation. (default: None) (int)</li>
<li><strong>use_signatures</strong> – Use a specific subset of all signatures. A list of signature names. (default: None) (list of str)</li>
<li><strong>ignore_signatures</strong> – Exclude a specific subset of all signatures from the analysis. A list of signature names. (default: None) (list of str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">The final proportions of all signatures in the mixture. (Filtered out or not used signatures appear with a proportion of zero.) (numpy.array)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.postprocess.get_details_for_mutations">
<code class="descclassname">isomut2py.postprocess.</code><code class="descname">get_details_for_mutations</code><span class="sig-paren">(</span><em>ref_fasta</em>, <em>input_dir</em>, <em>bam_filename</em>, <em>mutations_dataframe=None</em>, <em>output_dir=None</em>, <em>mutations_filename=None</em>, <em>samtools_fullpath='samtools'</em>, <em>base_quality_limit=30</em>, <em>samtools_flags=' -B -d 1000 '</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.postprocess.get_details_for_mutations" title="Permalink to this definition">¶</a></dt>
<dd><p>Get detailed results for the list of mutations contained in the mutations attribute of the object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>ref_fasta</strong> – the path to the reference genome fasta file (str)</li>
<li><strong>input_dir</strong> – the path to the directory where bam files are located (str)</li>
<li><strong>bam_filename</strong> – the filenames for the bam files of the investigated samples (list of str)</li>
<li><strong>mutations_dataframe</strong> – a pandas.DataFrame where mutations are stored (default: None) (pandas.DataFrame)</li>
<li><strong>output_dir</strong> – path to the directory where temporary files should be saved (default: ‘’) (str)</li>
<li><strong>mutations_filename</strong> – path to the file(s) where mutations are stored (default: None) (list of str)</li>
<li><strong>samtools_fullpath</strong> – path to samtools on the computer (default: “samtools’) (str)</li>
<li><strong>base_quality_limit</strong> – base quality limit for the pileup generation (default: 30) (int)</li>
<li><strong>samtools_flags</strong> – additional flags for samtools (default: ‘ -B -d ‘ + str(SAMTOOLS_MAX_DEPTH) + ‘ ‘) (str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">df_joined: A dataframe containing the detailed results. (pandas.DataFrame)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.postprocess.optimize_results">
<code class="descclassname">isomut2py.postprocess.</code><code class="descname">optimize_results</code><span class="sig-paren">(</span><em>sample_names</em>, <em>control_samples</em>, <em>FPs_per_genome</em>, <em>plot_roc=False</em>, <em>plot_tuning_curve=False</em>, <em>filtered_results_file=None</em>, <em>output_dir=None</em>, <em>mutations_dataframe=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.postprocess.optimize_results" title="Permalink to this definition">¶</a></dt>
<dd><p>Optimizes the list of detected mutations according to the list of control samples and desired level of false positives set by the user.
Filtered results will be loaded to the mutations attribute of the MutationDetection object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>sample_names</strong> – The list of sample names included in the analysis. (list of str)</li>
<li><strong>control_samples</strong> – List of sample names that should be used as control samples in the sense, that no unique mutations are expected in them. (The sample names listed here must match a subset of the sample names listed in bam_filename.) (list of str)</li>
<li><strong>FPs_per_genome</strong> – The total number of false positives tolerated in a control sample. (int)</li>
<li><strong>plot_roc</strong> – If True, ROC curves will be plotted as a visual representation of the optimization process. (default: False) (boolean)</li>
<li><strong>plot_tuning_curve</strong> – If True, tuning curves displaying the number of mutations found in different samples with different score filters will be plotted as a visual representation of the optimization process. (default: False) (boolean)</li>
<li><strong>filtered_results_file</strong> – The path to the file where filtered results should be saved. (default: [output_dir]/filtered_results.csv) (str)</li>
<li><strong>output_dir</strong> – the path to the directory where raw mutation tables are located (default: None) (str)</li>
<li><strong>mutations_dataframe</strong> – the pandas.DataFrame where mutations are located (default: None) (pandas.DataFrame)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last"><p>(score_lim_dict, filtered_results)</p>
<ul class="simple">
<li>score_lim_dict: a dictionary containing the optimized score values for each ploidy separately</li>
<li>filtered_results: a pandas.DataFrame containing the filtered mutations</li>
</ul>
</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.postprocess.optimize_score">
<code class="descclassname">isomut2py.postprocess.</code><code class="descname">optimize_score</code><span class="sig-paren">(</span><em>mutation_dataframe</em>, <em>control_samples</em>, <em>FPs_per_genome</em>, <em>score0=0</em>, <em>unique_samples=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.postprocess.optimize_score" title="Permalink to this definition">¶</a></dt>
<dd><p>Optimizes score values for different mutation types (SNV, INS, DEL) and ploidies according to the list of control samples and the desired
level of false positives in the genome. The results are stored in the score_lim_dict attribute of the MutationDetection object. If plot = True,
plots ROC curves for all mutations types (SNV, INS, DEL) and all ploidies.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>mutation_dataframe</strong> – The dataframe containing the mutations. (pandas.DataFrame)</li>
<li><strong>control_samples</strong> – a subset of bam_filename (list of sample names) that should be considered as control samples. Control samples are defined as samples where no unique mutations are expected to be found. (list of str)</li>
<li><strong>FPs_per_genome</strong> – the largest number of false positives tolerated in a control sample (int)</li>
<li><strong>score0</strong> – Score optimization starts with score0. If a larger score value is likely to be optimal, setting score0 to a number larger than 0 can decrease computation time. (default: 0) (float)</li>
<li><strong>unique_samples</strong> – list of unique samples where at least one mutation is detected (default: None) (list of str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">a dictionary containing the optimized score values for each ploidy</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="index.html">
              <img class="logo" src="_static/logo_Large.png" alt="Logo"/>
            </a></p>
<h1 class="logo"><a href="index.html">isomut2py</a></h1>



<p class="blurb">Comprehensive detection and analysis of genomic mutations from NGS sequencing data</p>






<h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="use_cases.html">General use cases</a></li>
<li class="toctree-l1"><a class="reference internal" href="external_mutations.html">Importing external mutations</a></li>
<li class="toctree-l1"><a class="reference internal" href="PE_advanced.html">Advanced ploidy estimation</a></li>
<li class="toctree-l1"><a class="reference internal" href="postprocessing.html">Further analysis, visualization</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="code.html">API reference</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="code_mutation_calling.html">MutationCaller object</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_ploidy_estimation.html">PloidyEstimator object</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_plotting_functions.html">Functions for plotting</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Functions for postprocessing lists of mutations</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_technical_functions.html">Technical (formatting, IO, etc.) functions</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  <li><a href="code.html">API reference</a><ul>
      <li>Previous: <a href="code_plotting_functions.html" title="previous chapter">Functions for plotting</a></li>
      <li>Next: <a href="code_technical_functions.html" title="next chapter">Technical (formatting, IO, etc.) functions</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2019, Orsolya Pipek.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.8.0</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.11</a>
      
      |
      <a href="_sources/code_postprocessing_functions.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>