
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Technical (formatting, IO, etc.) functions &#8212; isomut2py 2.0.1 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="_static/icon.ico"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Functions for postprocessing lists of mutations" href="code_postprocessing_functions.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="technical-formatting-io-etc-functions">
<h1>Technical (formatting, IO, etc.) functions<a class="headerlink" href="#technical-formatting-io-etc-functions" title="Permalink to this headline">¶</a></h1>
<p>Most of these functions can be called as class methods for <a class="reference internal" href="code_mutation_calling.html"><span class="doc">MutationCaller objects</span></a> and/or <a class="reference internal" href="code_ploidy_estimation.html"><span class="doc">PloidyEstimator objects</span></a>. They are described here individually, so that subtasks can be more easily managed.</p>
<div class="section" id="module-isomut2py.format">
<span id="formatting-functions"></span><h2>Formatting functions<a class="headerlink" href="#module-isomut2py.format" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="isomut2py.format.generate_ploidy_info_file">
<code class="descclassname">isomut2py.format.</code><code class="descname">generate_ploidy_info_file</code><span class="sig-paren">(</span><em>filename=None</em>, <em>sample_names=None</em>, <em>bed_filepaths=None</em>, <em>ploidy_estimation_objects=None</em>, <em>sample_groups=None</em>, <em>group_bed_filepaths=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.format.generate_ploidy_info_file" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Generate ploidy info file for mutation detection in samples with nondefault ploidies. Make sure to supply one of the following arguments:</dt>
<dd><ul class="first last simple">
<li>ploidy_estimation_objects</li>
<li>sample_names AND bed_filepaths</li>
<li>sample_groups AND group_bed_filepaths</li>
</ul>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>filename</strong> – The desired path to the generated ploidy info file. If None, ploidy information is saved to ploidy_info_file.txt in the current directory. (default: None) (str)</li>
<li><strong>sample_names</strong> – List of bam filenames for the samples, must be supplied together with bed_filepaths. (default: None) (list of str)</li>
<li><strong>bed_filepaths</strong> – Must be supplied together with sample_names. The list of filepaths to each bed file describing the given sample in sample_names. (default: None) (list of str)</li>
<li><strong>ploidy_estimation_objects</strong> – List of PloidyEstimation objects for each sample. (default: None) (list of isomut2py.PloidyEstimation)</li>
<li><strong>sample_groups</strong> – List of lists of str. Each list in sample_groups must contain the name of bam files in that group. Must be supplied together with group_bed_filepaths. (default: None) (list of list of str, example: [[‘sample1.bam’, ‘sample2.bam’, ‘sample3.bam’], [‘sample4.bam’, ‘sample5.bam’], [‘sample6.bam’]])</li>
<li><strong>group_bed_filepaths</strong> – List of filepaths to the bed files describing each sample group in sample_groups. Must be supplied together with sample_groups. (default: None) (list of str, example: [‘bedfile_of_samples123.txt’, ‘bedfile_of_samples45.txt’ ‘bedfile_of_samples6.txt’])</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.format.get_bed_format_for_sample">
<code class="descclassname">isomut2py.format.</code><code class="descname">get_bed_format_for_sample</code><span class="sig-paren">(</span><em>chromosomes</em>, <em>chrom_length</em>, <em>output_dir</em>, <em>bam_filename=None</em>, <em>ownbed_filepath=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.format.get_bed_format_for_sample" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates bed file of constant ploidies for a given sample from a file of positional ploidy data.
If the ownbed_filepath attribute of the PloidyEstimation object is set, saves the bedfile to the path specified there. Otherwise,
saves it to the output_dir with the “_ploidy.bed” suffix. Also sets the bed_dataframe attribute to the pandas.Dataframe containing the bed file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>chromosomes</strong> – list of chromosomes (array-like)</li>
<li><strong>chrom_length</strong> – list of chromosome lengths in bp (array-like)</li>
<li><strong>output_dir</strong> – path to the directory where the PE_fullchrom_* files are located. (str)</li>
<li><strong>bam_filename</strong> – filename of the BAM file of the sample (default: None) (str)</li>
<li><strong>ownbed_filepath</strong> – path to the bed file where results should be saved (default: None) (str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last"><p>(ownbed_filepath, df)</p>
<ul class="simple">
<li>ownbed_filepath: path the the bed file where results are saved</li>
<li>df: the bed file in a pandas.DataFrame</li>
</ul>
</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-isomut2py.io">
<span id="io-functions"></span><h2>IO functions<a class="headerlink" href="#module-isomut2py.io" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="isomut2py.io.get_coverage_distribution">
<code class="descclassname">isomut2py.io.</code><code class="descname">get_coverage_distribution</code><span class="sig-paren">(</span><em>chromosomes</em>, <em>output_dir</em>, <em>cov_max</em>, <em>cov_min</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.io.get_coverage_distribution" title="Permalink to this definition">¶</a></dt>
<dd><p>Sets the coverage_sample attribute of the PloidyEstimation object to the coverage distribution obtained from the temporary files created by
__PE_prepare_temp_files(). Positions are filtered according to the attributes of the PloidyEstimation object. The number of positions in the
final sample is decreased to 2000 for faster inference.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>chromosomes</strong> – list of chromosomes in the samples (array-like)</li>
<li><strong>output_dir</strong> – path to the directory where temporary files (PEtmp_fullchrom*) are located (str)</li>
<li><strong>cov_max</strong> – the maximum value of the coverage for a position to be included in the analysis (int)</li>
<li><strong>cov_min</strong> – the minimum value of the coverage for a position to be included in the analysis (int)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">a 2000-element sample from the coverage distribution</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.io.load_bedfile_from_file">
<code class="descclassname">isomut2py.io.</code><code class="descname">load_bedfile_from_file</code><span class="sig-paren">(</span><em>filename=None</em>, <em>output_dir=None</em>, <em>bam_filename=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.io.load_bedfile_from_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the bedfile containing previous ploidy estimated for the given sample from the path specified in filename.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>filename</strong> – The path to the bedfile. If not supplied, the bed file is attempted to be loaded from [output_dir]/[bam_filename]_ploidy.bed. (default: None) (str)</li>
<li><strong>output_dir</strong> – The path to the directory where the default bedfile is located. (default: None) (str)</li>
<li><strong>bam_filename</strong> – the filename of the BAM file of the sample (default: None) (str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">the pandas DataFrame of the bedfile</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.io.load_cov_distribution_parameters_from_file">
<code class="descclassname">isomut2py.io.</code><code class="descname">load_cov_distribution_parameters_from_file</code><span class="sig-paren">(</span><em>filename=None</em>, <em>output_dir=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.io.load_cov_distribution_parameters_from_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the parameters of the seven fitted Gaussians to the coverage distribution of the sample from the specified filename (that was saved with
pickle beforehand). If one such file is available, the computationally expensive ploidy estimation process can be skipped.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>filename</strong> – The path to the file with the coverage distribution parameters. (default: None) (str)</li>
<li><strong>output_dir</strong> – if filename is not supplied, distribution parameters are attempted to be loaded from [output_dir]/GaussDistParams.pkl (default: None) (str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">dictionary containing the fitted coverage distribution parameters</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.io.load_mutations">
<code class="descclassname">isomut2py.io.</code><code class="descname">load_mutations</code><span class="sig-paren">(</span><em>output_dir</em>, <em>filename=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.io.load_mutations" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads mutations from a file or a list of files.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>filename</strong> – The path to the file, where mutations are stored. A list of paths can be also supplied, in this case, all of them will be loaded to a single dataframe. If None, the file [output_dir]/filtered_mutations.csv will be loaded. (default: None) (str)</li>
<li><strong>output_dir</strong> – The path to the directory containing the file filtered_results.csv. (str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">a pandas.DataFrame containing the mutations</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.io.load_obj">
<code class="descclassname">isomut2py.io.</code><code class="descname">load_obj</code><span class="sig-paren">(</span><em>name</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.io.load_obj" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads a python object from a file with the name (name).pkl.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>name</strong> – filename (without extension)</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">loaded object</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.io.save_obj">
<code class="descclassname">isomut2py.io.</code><code class="descname">save_obj</code><span class="sig-paren">(</span><em>obj</em>, <em>name</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.io.save_obj" title="Permalink to this definition">¶</a></dt>
<dd><p>Saves a python object (obj) with the specified name as (name).pkl.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>obj</strong> – object to save</li>
<li><strong>name</strong> – filename (without extension)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="processing-functions">
<h2>Processing functions<a class="headerlink" href="#processing-functions" title="Permalink to this headline">¶</a></h2>
<p>(These functions mainly take care of parallelization.)</p>
<span class="target" id="module-isomut2py.process"></span><dl class="function">
<dt id="isomut2py.process.PE_prepare_temp_files">
<code class="descclassname">isomut2py.process.</code><code class="descname">PE_prepare_temp_files</code><span class="sig-paren">(</span><em>ref_fasta</em>, <em>input_dir</em>, <em>bam_filename</em>, <em>output_dir</em>, <em>genome_length</em>, <em>n_min_block</em>, <em>n_conc_blocks</em>, <em>chromosomes</em>, <em>chrom_length</em>, <em>windowsize=10000</em>, <em>shiftsize=3000</em>, <em>min_noise=0.1</em>, <em>print_every_nth=1000</em>, <em>base_quality_limit=0</em>, <em>samtools_fullpath='samtools'</em>, <em>samtools_flags=None</em>, <em>bedfile=None</em>, <em>level=0</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.process.PE_prepare_temp_files" title="Permalink to this definition">¶</a></dt>
<dd><p>Prepares temporary files for ploidy estimation, by averaging coverage in moving windows for the whole genome and collecting positions with
reference allele frequencies in the [min_noise, 1-min_noise] range.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>ref_fasta</strong> – path to the reference genome fasta file (str)</li>
<li><strong>input_dir</strong> – path to the folder where bam files are located (str)</li>
<li><strong>bam_filename</strong> – list of bam filenames to analyse (list of str)</li>
<li><strong>output_dir</strong> – path to the directory where results should be saved (str)</li>
<li><strong>genome_length</strong> – the total length of the genome in basepairs (int)</li>
<li><strong>n_min_block</strong> – The approximate number of blocks to partition the analysed genome to for parallel computing. The actual number might be slightly larger that this. (default: 200) (int)</li>
<li><strong>n_conc_blocks</strong> – The number of blocks to process at the same time. (default: 4) (int)</li>
<li><strong>chromosomes</strong> – list of chromosomes in the genome (list of str)</li>
<li><strong>chrom_length</strong> – list of chromosome lengths in the genome (list of int)</li>
<li><strong>windowsize</strong> – windowsize for initial coverage smoothing with a moving average method (default: 10000) (int)</li>
<li><strong>shiftsize</strong> – shiftsize for initial coverage smoothing with a moving average method (default: 3000) (int)</li>
<li><strong>min_noise</strong> – The minimum frequency of non-reference or reference bases detected for a position to be considered for LOH detection. Setting it too small will result in poor noise filtering, setting it too large will result in a decreased number of measurement points. (default: 0.1) (float in range(0,1))</li>
<li><strong>print_every_nth</strong> – Even though LOH detection is limited to the positions with a noise level larger that min_noise, ploidy estimation is based on all the genomic positions meeting the above set criteria. By setting the attribute print_every_nth, the number of positions used can be controlled. Setting it large will result in overlooking ploidy variations in shorter genomic ranges, while setting it too small can cause an increase in both memory usage and computation time. Decrease only if a relatively short genome is analysed. (default: 100) (int)</li>
<li><strong>base_quality_limit</strong> – The base quality limit used by samtools in order to decide if a base should be included in the pileup file. (default: 0) (int)</li>
<li><strong>samtools_fullpath</strong> – The path to samtools on the computer. (default: “samtools”) (str)</li>
<li><strong>samtools_flags</strong> – The samtools flags to be used for pileup file generation. (default: ” -B -d 1000 “) (str)</li>
<li><strong>bedfile</strong> – path to the the bedfile to limit ploidy estimation to a specific region of the genome (default: None) (str)</li>
<li><strong>level</strong> – the level of indentation used in verbose output (default: 0) (int)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.process.double_run">
<code class="descclassname">isomut2py.process.</code><code class="descname">double_run</code><span class="sig-paren">(</span><em>mutcallObject</em>, <em>level=0</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.process.double_run" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs the mutation detection pipeline on the MutationCaller object when with local realignment.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>mutcallObject</strong> – an isomut2py.MutationCaller object</li>
<li><strong>level</strong> – the level of indentation used in verbose output (default: 0) (int)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.process.single_run">
<code class="descclassname">isomut2py.process.</code><code class="descname">single_run</code><span class="sig-paren">(</span><em>mutcallObject</em>, <em>level=0</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.process.single_run" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs the mutation detection pipeline on the MutationCaller object when no local realignment is used during the process.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>mutcallObject</strong> – an isomut2py.MutationCaller object</li>
<li><strong>level</strong> – the level of indentation used in verbose output (default: 0) (int)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="bayesian-inference-functions">
<h2>Bayesian inference functions<a class="headerlink" href="#bayesian-inference-functions" title="Permalink to this headline">¶</a></h2>
<p>(These functions are called by the <a class="reference internal" href="code_ploidy_estimation.html"><span class="doc">PloidyEstimator object</span></a> to fit different theoretical distributions to the actual coverage distribution calculated from the data.)</p>
<span class="target" id="module-isomut2py.bayesian"></span><dl class="function">
<dt id="isomut2py.bayesian.PE_on_chrom">
<code class="descclassname">isomut2py.bayesian.</code><code class="descname">PE_on_chrom</code><span class="sig-paren">(</span><em>chrom</em>, <em>output_dir</em>, <em>windowsize_PE</em>, <em>shiftsize_PE</em>, <em>distribution_dict</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.bayesian.PE_on_chrom" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs the whole ploidy estimation pipeline on a given chromosome, using the appropriate attributes of the PloidyEstimation object by running
PE_on_range() multiple times.
Prints the results to the file: [output_dir]/PE_fullchrom_[chrom].txt.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>distribution_dict</strong> – a dictionary containing the fitted parameters of the Gaussian mixture model to the coverage distribution. (dict with keys ‘mu’, ‘sigma’ and ‘p’)</li>
<li><strong>shiftsize_PE</strong> – shiftsize for moving average over regions (int)</li>
<li><strong>windowsize_PE</strong> – windowsize for moving average over regions (int)</li>
<li><strong>output_dir</strong> – the path to the directory where temporary files are located (str)</li>
<li><strong>chrom</strong> – the name of the chromosome to analyse (str)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.bayesian.PE_on_range">
<code class="descclassname">isomut2py.bayesian.</code><code class="descname">PE_on_range</code><span class="sig-paren">(</span><em>dataframe</em>, <em>rmin</em>, <em>rmax</em>, <em>all_mu</em>, <em>all_sigma</em>, <em>prior</em>, <em>cov_min=0</em>, <em>cov_max=100000</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.bayesian.PE_on_range" title="Permalink to this definition">¶</a></dt>
<dd><p>Run the ploidy estimation on a given range of a chromosome.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>dataframe</strong> – The dataframe read from the temporary files for a given chromosome, containing information about the genomic position, the measured coverage and the frequency of the non-reference bases aligned to the position. (pandas.DataFrame)</li>
<li><strong>rmin</strong> – The lower bound of the genomic range considered for the analysis. (int)</li>
<li><strong>rmax</strong> – The upper bound of the genomic range considered for the analysis. (int)</li>
<li><strong>all_mu</strong> – The centers of the seven Gaussians fitted to the coverage distribution. (list of float)</li>
<li><strong>all_sigma</strong> – The sigmas of the seven Gaussians fitted to the coverage distribution. (list of float)</li>
<li><strong>prior</strong> – The weights of the seven Gaussians fitted to the coverage distribution. (list of float in the range(0,1))</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last"><p>(most_probable_ploidy, most_probable_LOH)</p>
<ul class="simple">
<li>most_probable_ploidy: the estimated ploidy for the genomic region (int, in range(1,8))</li>
<li>most_probable_LOH: the estimated LOH status for the genomic region (1: LOH, 0: no LOH)</li>
</ul>
</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.bayesian.estimate_hapcov_infmix">
<code class="descclassname">isomut2py.bayesian.</code><code class="descname">estimate_hapcov_infmix</code><span class="sig-paren">(</span><em>cov_min=None</em>, <em>hc_percentile=None</em>, <em>chromosomes=None</em>, <em>output_dir=None</em>, <em>cov_max=None</em>, <em>level=0</em>, <em>cov_sample=None</em>, <em>user_defined_hapcov=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.bayesian.estimate_hapcov_infmix" title="Permalink to this definition">¶</a></dt>
<dd><p>Estimates the haploid coverage of the sample.
If the user_defined_hapcov attribute is set manually, it sets the value of estimated_hapcov to that.
Otherwise, a many-component (20) Gaussian mixture model is fitted to the coverage histogram of the sample 10 times. Each time,
the haploid coverage is estimated from the center of the component with the maximal weight in the model. The final estimate of the haploid
coverage is calculated as the qth percentile of the 10 measurements, with q = hc_percentile.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>user_defined_hapcov</strong> – if not None, its value is returned (default: None) (float)</li>
<li><strong>cov_sample</strong> – a sample of the coverage distribution of the investigated sample, if None, it is loaded from the temporary files of the output_dir (default: None) (array-like)</li>
<li><strong>cov_max</strong> – the maximum value of the coverage for a position to be considered in the estimation (default: None) (int)</li>
<li><strong>output_dir</strong> – the path to the output directory of the PloidyEstimator object, where temporary files are located (default: None) (str)</li>
<li><strong>chromosomes</strong> – list of chromosomes for the sample (default: None) (array-like)</li>
<li><strong>hc_percentile</strong> – the percentile value to use for calculating the estimated haploid coverage from 10 subsequent estimations (default: None) (int)</li>
<li><strong>cov_min</strong> – the maximum value of the coverage for a position to be considered in the estimation (default: None) (int)</li>
<li><strong>level</strong> – the level of indentation used in verbose output (default: 0) (int)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">(estimated haploid coverage (float), sample from coverage distribution (array-like))</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.bayesian.fit_gaussians">
<code class="descclassname">isomut2py.bayesian.</code><code class="descname">fit_gaussians</code><span class="sig-paren">(</span><em>estimated_hapcov</em>, <em>chromosomes=None</em>, <em>output_dir=None</em>, <em>cov_max=None</em>, <em>cov_min=None</em>, <em>level=0</em>, <em>cov_sample=None</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.bayesian.fit_gaussians" title="Permalink to this definition">¶</a></dt>
<dd><p>Fits a 7-component Gaussian mixture model to the coverage distribution of the sample, using the appropriate attributes of the PloidyEstimation
object. The center of the first Gaussian is initialized from a narrow region around the value of the estimated_hapcov attribute. The centers of
the other Gaussians are initialized in a region around the value of estimated_hapcov multiplied by consecutive whole numbers.</p>
<p>The parameters of the fitted model (center, sigma and weight) for all seven Gaussians are both saved to the GaussDistParams.pkl file (in
output_dir, for later reuse) and set as the value of the distribution_dict attribute.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>cov_sample</strong> – a sample of the coverage distribution of the investigated sample, if None, it is loaded from the temporary files of the output_dir (default: None) (array-like)</li>
<li><strong>cov_min</strong> – the maximum value of the coverage for a position to be considered in the estimation (default: None) (int)</li>
<li><strong>output_dir</strong> – the path to the output directory of the PloidyEstimator object, where temporary files are located. If not None, distribution parameters are saved there as GaussDistParams.pkl. (default: None) (str)</li>
<li><strong>chromosomes</strong> – list of chromosomes for the sample (default: None) (array-like)</li>
<li><strong>estimated_hapcov</strong> – the estimated value for the haploid coverage, used as prior (float)</li>
<li><strong>level</strong> – the level of indentation used in verbose output (default: 0) (int)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">dictionary containing the fitted parameters of the 7 Gaussians</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="functions-for-ploidy-comparison">
<h2>Functions for ploidy comparison<a class="headerlink" href="#functions-for-ploidy-comparison" title="Permalink to this headline">¶</a></h2>
<p>(These functions perform the comparison of ploidy estimates of two samples for different file formats.)</p>
<span class="target" id="module-isomut2py.compare"></span><dl class="function">
<dt id="isomut2py.compare.check_interval_for_difference">
<code class="descclassname">isomut2py.compare.</code><code class="descname">check_interval_for_difference</code><span class="sig-paren">(</span><em>chrom</em>, <em>chromStart</em>, <em>chromEnd</em>, <em>ploidy1</em>, <em>ploidy2</em>, <em>original_bamfile1</em>, <em>original_bamfile2</em>, <em>refgenome</em>, <em>prior_dist_dict1</em>, <em>prior_dist_dict2</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.compare.check_interval_for_difference" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if a genomic interval with different estimated ploidies is really different in the two samples.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>chrom</strong> – The chromosome of the genomic interval. (str)</li>
<li><strong>chromStart</strong> – The starting position of the genomic interval. (int)</li>
<li><strong>chromEnd</strong> – The ending position of the genomic interval. (int)</li>
<li><strong>ploidy1</strong> – The ploidy estimated for the genomic interval for original_bamfile1. (int)</li>
<li><strong>ploidy2</strong> – The ploidy estimated for the genomic interval for original_bamfile2. (int)</li>
<li><strong>original_bamfile1</strong> – The path to the original bamfile containing alignment information for sample1. (str)</li>
<li><strong>original_bamfile2</strong> – The path to the original bamfile containing alignment information for sample2. (str)</li>
<li><strong>refgenome</strong> – The path to the reference genome fasta file. (str)</li>
<li><strong>prior_dist_dict1</strong> – The parameters of the seven Gaussians fitted to the coverage distribution of sample1. (dict)</li>
<li><strong>prior_dist_dict2</strong> – The parameters of the seven Gaussians fitted to the coverage distribution of sample2. (dict)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">the ratio of the likelihood of the two samples having different ploidies in the region and the likelihood of them having the same ploidies (float)
- if the ratio of them having different ploidies is smaller than the ratio of them having the same one, the returned value is 0</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.compare.compare_with_bed">
<code class="descclassname">isomut2py.compare.</code><code class="descname">compare_with_bed</code><span class="sig-paren">(</span><em>bed_dataframe</em>, <em>other_file</em>, <em>minLen</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.compare.compare_with_bed" title="Permalink to this definition">¶</a></dt>
<dd><p>Compares the results of ploidy estimation with a bed file defined in other_file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>bed_dataframe</strong> – a pandas.DataFrame of the bedfile of the sample (pandas.DataFrame)</li>
<li><strong>other_file</strong> – The path to the bedfile of the other sample. (str)</li>
<li><strong>minLen</strong> – The minimum length of a region to be considered different from the other_file. (int)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">df_joined: A pandas.DataFrame containing region information from both the PloidyEstimation object and the other_file.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.compare.compare_with_other_PloidyEstimator">
<code class="descclassname">isomut2py.compare.</code><code class="descname">compare_with_other_PloidyEstimator</code><span class="sig-paren">(</span><em>ob1</em>, <em>ob2</em>, <em>minLen</em>, <em>minQual</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.compare.compare_with_other_PloidyEstimator" title="Permalink to this definition">¶</a></dt>
<dd><p>Compare the estimated ploidies of the PloidyEstimation object with the ploidies of another PloidyEstimator object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>ob2</strong> – the other PloidyEstimator object (isomut2py.ploidyestimation.PloidyEstimator)</li>
<li><strong>ob1</strong> – the first PloidyEstimator object (isomut2py.ploidyestimation.PloidyEstimator)</li>
<li><strong>minLen</strong> – The minimum length of a region to be considered different from the other object. (int)</li>
<li><strong>minQual</strong> – The minimum quality of a region to be considered different from the other object. (float)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">df_intervals: The differing intervals meeting the above criteria. (pandas.DataFrame)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="functions-for-loading-example-parameter-settings">
<h2>Functions for loading example parameter settings<a class="headerlink" href="#functions-for-loading-example-parameter-settings" title="Permalink to this headline">¶</a></h2>
<p>(These functions download example datasets and help load the settings for processing these in a concise way.)</p>
<span class="target" id="module-isomut2py.examples"></span><dl class="function">
<dt id="isomut2py.examples.download_example_data">
<code class="descclassname">isomut2py.examples.</code><code class="descname">download_example_data</code><span class="sig-paren">(</span><em>path='.'</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.examples.download_example_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Download example data from <a class="reference external" href="http://genomics.hu/tools/isomut2py/isomut2py_exampleDataset.tar.gz">http://genomics.hu/tools/isomut2py/isomut2py_exampleDataset.tar.gz</a> to path.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>path</strong> – where to download (default: ‘.’) (str)</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.examples.download_raw_example_data">
<code class="descclassname">isomut2py.examples.</code><code class="descname">download_raw_example_data</code><span class="sig-paren">(</span><em>path='.'</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.examples.download_raw_example_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Download raw example data from either <a class="reference external" href="http://genomics.hu/tools/isomut2py/isomut2py_rawExampleDataset.tar.gz">http://genomics.hu/tools/isomut2py/isomut2py_rawExampleDataset.tar.gz</a> or <a class="reference external" href="http://genomics.hu/tools/isomut2py/isomut2py_rawExampleDataset_shortgenome.tar.gz">http://genomics.hu/tools/isomut2py/isomut2py_rawExampleDataset_shortgenome.tar.gz</a> to path.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>path</strong> – where to download (default: ‘.’) (str)</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.examples.load_example_mutdet_settings">
<code class="descclassname">isomut2py.examples.</code><code class="descname">load_example_mutdet_settings</code><span class="sig-paren">(</span><em>example_data_path='.'</em>, <em>output_dir='.'</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.examples.load_example_mutdet_settings" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads example settings for mutation detection.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>example_data_path</strong> – the path where example data is located (default: ‘.’) (str)</li>
<li><strong>output_dir</strong> – the path where results should be saved (default: current working directory) (str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">dictionary containing example parameters</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.examples.load_example_ploidyest_settings">
<code class="descclassname">isomut2py.examples.</code><code class="descname">load_example_ploidyest_settings</code><span class="sig-paren">(</span><em>example_data_path='.'</em>, <em>output_dir='.'</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.examples.load_example_ploidyest_settings" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads example settings for ploidy estimation.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>example_data_path</strong> – the path where example data is located (default: ‘.’) (str)</li>
<li><strong>output_dir</strong> – the path where results should be saved (default: current working directory) (str)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">dictionary containing example parameters</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="isomut2py.examples.load_preprocessed_example_ploidyest_settings">
<code class="descclassname">isomut2py.examples.</code><code class="descname">load_preprocessed_example_ploidyest_settings</code><span class="sig-paren">(</span><em>example_data_path='.'</em><span class="sig-paren">)</span><a class="headerlink" href="#isomut2py.examples.load_preprocessed_example_ploidyest_settings" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads example settings for ploidy estimation with preprocessed files.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>example_data_path</strong> – the path where example data is located (default: ‘.’) (str)</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">dictionary containing example parameters</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="index.html">
              <img class="logo" src="_static/logo_Large.png" alt="Logo"/>
            </a></p>
<h1 class="logo"><a href="index.html">isomut2py</a></h1>



<p class="blurb">Comprehensive detection and analysis of genomic mutations from NGS sequencing data</p>






<h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="use_cases.html">General use cases</a></li>
<li class="toctree-l1"><a class="reference internal" href="external_mutations.html">Importing external mutations</a></li>
<li class="toctree-l1"><a class="reference internal" href="PE_advanced.html">Advanced ploidy estimation</a></li>
<li class="toctree-l1"><a class="reference internal" href="postprocessing.html">Further analysis, visualization</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="code.html">API reference</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="code_mutation_calling.html">MutationCaller object</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_ploidy_estimation.html">PloidyEstimator object</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_plotting_functions.html">Functions for plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_postprocessing_functions.html">Functions for postprocessing lists of mutations</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Technical (formatting, IO, etc.) functions</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  <li><a href="code.html">API reference</a><ul>
      <li>Previous: <a href="code_postprocessing_functions.html" title="previous chapter">Functions for postprocessing lists of mutations</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2019, Orsolya Pipek.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.8.0</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.11</a>
      
      |
      <a href="_sources/code_technical_functions.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>