
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Importing external mutations &#8212; isomut2py 2.0.1 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="_static/icon.ico"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Advanced ploidy estimation" href="PE_advanced.html" />
    <link rel="prev" title="General use cases" href="use_cases.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #303F9F;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #D84315;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 8ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #cfcfcf;
    border-radius: 2px;
    background: #f7f7f7;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.pngmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }
</style>
<div class="section" id="Importing-external-mutations">
<h1>Importing external mutations<a class="headerlink" href="#Importing-external-mutations" title="Permalink to this headline">¶</a></h1>
<p>For instructions on installation and basic exploration steps, see
<a class="reference internal" href="getting_started.html"><span class="doc">Getting started</span></a>.</p>
<p>It is possible to post-process lists of mutations with <code class="docutils literal notranslate"><span class="pre">isomut2py</span></code>
that were otherwise generated with an external variant caller tool. Here
we demonstrate how external VCF files can be loaded and further
analysed.</p>
<p>As <code class="docutils literal notranslate"><span class="pre">isomut2py</span></code> expects <code class="docutils literal notranslate"><span class="pre">pandas</span></code> DataFrames as inputs to handle
mutation lists, we will be using the <code class="docutils literal notranslate"><span class="pre">pandas</span></code> package for the loading
of VCF files.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">isomut2py</span> <span class="k">as</span> <span class="nn">im2</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="o">%</span><span class="k">matplotlib</span> inline
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Compiling C scripts...
Done.
</pre></div></div>
</div>
<div class="section" id="Importing-VCF-files-using-pyvcf">
<h2>Importing VCF files using pyvcf<a class="headerlink" href="#Importing-VCF-files-using-pyvcf" title="Permalink to this headline">¶</a></h2>
<p>VCF files can be easily processed in python with the <code class="docutils literal notranslate"><span class="pre">pyvcf</span></code> package.
This can be installed with:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">pip</span> <span class="n">install</span> <span class="n">pyvcf</span>
</pre></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">vcf</span>
</pre></div>
</div>
</div>
<p>To convert VCF files to <code class="docutils literal notranslate"><span class="pre">pandas</span></code> DataFrames with columns that can be
parsed by <code class="docutils literal notranslate"><span class="pre">isomut2py</span></code>, we first need to make sure to find the values
of <code class="docutils literal notranslate"><span class="pre">sample_name</span></code>, <code class="docutils literal notranslate"><span class="pre">chr</span></code>, <code class="docutils literal notranslate"><span class="pre">pos</span></code>, <code class="docutils literal notranslate"><span class="pre">type</span></code>, <code class="docutils literal notranslate"><span class="pre">score</span></code>, <code class="docutils literal notranslate"><span class="pre">ref</span></code>,
<code class="docutils literal notranslate"><span class="pre">mut</span></code>, <code class="docutils literal notranslate"><span class="pre">cov</span></code>, <code class="docutils literal notranslate"><span class="pre">mut_freq</span></code>, <code class="docutils literal notranslate"><span class="pre">cleanliness</span></code> and <code class="docutils literal notranslate"><span class="pre">ploidy</span></code>. In order
to perform downstream analysis of mutations, fields <code class="docutils literal notranslate"><span class="pre">cov</span></code>,
<code class="docutils literal notranslate"><span class="pre">mut_freq</span></code> and <code class="docutils literal notranslate"><span class="pre">cleanliness</span></code> can be left empty, but nonetheless have
to be defined in the dataframe.</p>
<p>Here we are importing VCF files generated by the tool MuTect2 (of GATK).
If another tool was used for variant calling, make sure to modify parser
function below. The example VCF files are located at
<code class="docutils literal notranslate"><span class="pre">[exampleDataDir]/isomut2py_example_dataset/ExternalMutations/mutect2</span></code>.
(For instructions on how to download the example datafiles, see <a class="reference internal" href="getting_started.html"><span class="doc">Getting
started</span></a>.)</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [91]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="k">def</span> <span class="nf">parse_VCF_record</span><span class="p">(</span><span class="n">record</span><span class="p">):</span>
    <span class="k">if</span> <span class="n">record</span><span class="o">.</span><span class="n">is_deletion</span><span class="p">:</span>
        <span class="n">muttype</span> <span class="o">=</span> <span class="s1">&#39;DEL&#39;</span>
    <span class="k">elif</span> <span class="n">record</span><span class="o">.</span><span class="n">is_indel</span><span class="p">:</span>
        <span class="n">muttype</span> <span class="o">=</span> <span class="s1">&#39;INS&#39;</span>
    <span class="k">elif</span> <span class="n">record</span><span class="o">.</span><span class="n">is_snp</span><span class="p">:</span>
        <span class="n">muttype</span> <span class="o">=</span> <span class="s1">&#39;SNV&#39;</span>

    <span class="k">return</span> <span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">CHROM</span><span class="p">,</span> <span class="n">record</span><span class="o">.</span><span class="n">POS</span><span class="p">,</span> <span class="n">muttype</span><span class="p">,</span>
            <span class="n">record</span><span class="o">.</span><span class="n">INFO</span><span class="p">[</span><span class="s1">&#39;TLOD&#39;</span><span class="p">][</span><span class="mi">0</span><span class="p">],</span>
            <span class="n">record</span><span class="o">.</span><span class="n">REF</span><span class="p">,</span> <span class="n">record</span><span class="o">.</span><span class="n">ALT</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span>
            <span class="nb">int</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">samples</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">DP</span><span class="p">),</span>
            <span class="nb">int</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">samples</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">AD</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span><span class="o">/</span><span class="nb">int</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">samples</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">DP</span><span class="p">),</span>
            <span class="nb">int</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">samples</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">AD</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span><span class="o">/</span><span class="nb">int</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">samples</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">DP</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [108]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">sample_dataframes</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">7</span><span class="p">):</span>
    <span class="n">vcf_reader</span> <span class="o">=</span> <span class="n">vcf</span><span class="o">.</span><span class="n">Reader</span><span class="p">(</span><span class="n">filename</span> <span class="o">=</span> <span class="n">exampleDataDir</span> <span class="o">+</span> <span class="s1">&#39;isomut2py_example_dataset/ExternalMutations/mutect2/&#39;</span><span class="o">+</span><span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">)</span><span class="o">+</span><span class="s1">&#39;_somatic_m.vcf.gz&#39;</span><span class="p">)</span>
    <span class="n">d</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">record</span> <span class="ow">in</span> <span class="n">vcf_reader</span><span class="p">:</span>
        <span class="n">d</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">parse_VCF_record</span><span class="p">(</span><span class="n">record</span><span class="p">))</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">d</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;chr&#39;</span><span class="p">,</span> <span class="s1">&#39;pos&#39;</span><span class="p">,</span> <span class="s1">&#39;type&#39;</span><span class="p">,</span> <span class="s1">&#39;score&#39;</span><span class="p">,</span> <span class="s1">&#39;ref&#39;</span><span class="p">,</span> <span class="s1">&#39;mut&#39;</span><span class="p">,</span> <span class="s1">&#39;cov&#39;</span><span class="p">,</span> <span class="s1">&#39;mut_freq&#39;</span><span class="p">,</span> <span class="s1">&#39;cleanliness&#39;</span><span class="p">])</span>
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;sample_name&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;sample_&#39;</span><span class="o">+</span><span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">)</span>
    <span class="n">df</span><span class="p">[</span><span class="s1">&#39;ploidy&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="mi">2</span>
    <span class="n">sample_dataframes</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">df</span><span class="p">[[</span><span class="s1">&#39;sample_name&#39;</span><span class="p">,</span><span class="s1">&#39;chr&#39;</span><span class="p">,</span> <span class="s1">&#39;pos&#39;</span><span class="p">,</span> <span class="s1">&#39;type&#39;</span><span class="p">,</span> <span class="s1">&#39;score&#39;</span><span class="p">,</span> <span class="s1">&#39;ref&#39;</span><span class="p">,</span>
                                <span class="s1">&#39;mut&#39;</span><span class="p">,</span> <span class="s1">&#39;cov&#39;</span><span class="p">,</span> <span class="s1">&#39;mut_freq&#39;</span><span class="p">,</span> <span class="s1">&#39;cleanliness&#39;</span><span class="p">,</span> <span class="s1">&#39;ploidy&#39;</span><span class="p">]])</span>

<span class="n">mutations_dataframe</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">sample_dataframes</span><span class="p">)</span>
</pre></div>
</div>
</div>
<p>Now we have all the mutations found in 6 samples in a single dataframe.
This can be further analysed with the functions described in <a class="reference internal" href="postprocessing.html"><span class="doc">Further
analysis, visualization</span></a>. For example the
number of mutations found in each sample can be plotted with:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [112]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">f</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">plot_mutation_counts</span><span class="p">(</span><span class="n">mutations_dataframe</span><span class="o">=</span><span class="n">mutations_dataframe</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Warning: list of control samples not defined.
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/external_mutations_9_1.png" src="_images/external_mutations_9_1.png" />
</div>
</div>
</div>
<div class="section" id="Importing-lists-of-mutations-in-arbitrary-files">
<h2>Importing lists of mutations in arbitrary files<a class="headerlink" href="#Importing-lists-of-mutations-in-arbitrary-files" title="Permalink to this headline">¶</a></h2>
<p>If your lists of mutations are stored in some kind of text files as
tables, the easiest way to import them is to use the <code class="docutils literal notranslate"><span class="pre">pandas</span></code> python
package. (As <code class="docutils literal notranslate"><span class="pre">isomut2py</span></code> heavily relies on <code class="docutils literal notranslate"><span class="pre">pandas</span></code>, it should
already be installed on your computer.)</p>
<p>Here we merely include an example for such an import, make sure to
modify the code and customize it to your table. (This data was generated
with MuTect, that only detects SNPs, thus we don’t expect to see any
indels.)</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">exampleDataDir</span> <span class="o">=</span> <span class="s1">&#39;/nagyvinyok/adat83/sotejedlik/orsi/&#39;</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [27]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">sample_dataframes</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">7</span><span class="p">):</span>
    <span class="k">for</span> <span class="n">sample_type</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;N&#39;</span><span class="p">,</span> <span class="s1">&#39;T&#39;</span><span class="p">]:</span>
        <span class="n">df_sample</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">exampleDataDir</span> <span class="o">+</span> <span class="s1">&#39;isomut2py_example_dataset/ExternalMutations/mutect/SV0&#39;</span><span class="o">+</span>
                                <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">)</span><span class="o">+</span><span class="n">sample_type</span><span class="o">+</span><span class="s1">&#39;_chr1.vcf&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">comment</span><span class="o">=</span><span class="s1">&#39;#&#39;</span><span class="p">)</span>

        <span class="n">df_sample</span> <span class="o">=</span> <span class="n">df_sample</span><span class="p">[</span><span class="n">df_sample</span><span class="p">[</span><span class="s1">&#39;judgement&#39;</span><span class="p">]</span> <span class="o">!=</span> <span class="s1">&#39;REJECT&#39;</span><span class="p">]</span>
        <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">()</span>
        <span class="n">df</span><span class="p">[</span><span class="s1">&#39;chr&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df_sample</span><span class="p">[</span><span class="s1">&#39;contig&#39;</span><span class="p">]</span>
        <span class="n">df</span><span class="p">[</span><span class="s1">&#39;pos&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df_sample</span><span class="p">[</span><span class="s1">&#39;position&#39;</span><span class="p">]</span>
        <span class="n">df</span><span class="p">[</span><span class="s1">&#39;type&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;SNV&#39;</span>
        <span class="n">df</span><span class="p">[</span><span class="s1">&#39;score&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df_sample</span><span class="p">[</span><span class="s1">&#39;t_lod_fstar&#39;</span><span class="p">]</span>
        <span class="n">df</span><span class="p">[</span><span class="s1">&#39;ref&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df_sample</span><span class="p">[</span><span class="s1">&#39;ref_allele&#39;</span><span class="p">]</span>
        <span class="n">df</span><span class="p">[</span><span class="s1">&#39;mut&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df_sample</span><span class="p">[</span><span class="s1">&#39;alt_allele&#39;</span><span class="p">]</span>
        <span class="n">df</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df_sample</span><span class="p">[</span><span class="s1">&#39;t_ref_count&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">df_sample</span><span class="p">[</span><span class="s1">&#39;t_alt_count&#39;</span><span class="p">]</span>
        <span class="n">df</span><span class="p">[</span><span class="s1">&#39;mut_freq&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df_sample</span><span class="p">[</span><span class="s1">&#39;t_alt_count&#39;</span><span class="p">]</span><span class="o">/</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;cov&#39;</span><span class="p">]</span>
        <span class="n">df</span><span class="p">[</span><span class="s1">&#39;cleanliness&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">df_sample</span><span class="p">[</span><span class="s1">&#39;n_ref_count&#39;</span><span class="p">]</span><span class="o">/</span><span class="p">(</span><span class="n">df_sample</span><span class="p">[</span><span class="s1">&#39;n_ref_count&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">df_sample</span><span class="p">[</span><span class="s1">&#39;n_alt_count&#39;</span><span class="p">])</span>
        <span class="n">df</span><span class="p">[</span><span class="s1">&#39;ploidy&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="mi">2</span>
        <span class="n">df</span><span class="p">[</span><span class="s1">&#39;sample_name&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;SV0&#39;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">i</span><span class="p">)</span> <span class="o">+</span> <span class="n">sample_type</span>
        <span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="s1">&#39;sample_name&#39;</span><span class="p">,</span>
         <span class="s1">&#39;chr&#39;</span><span class="p">,</span>
         <span class="s1">&#39;pos&#39;</span><span class="p">,</span>
         <span class="s1">&#39;type&#39;</span><span class="p">,</span>
         <span class="s1">&#39;score&#39;</span><span class="p">,</span>
         <span class="s1">&#39;ref&#39;</span><span class="p">,</span>
         <span class="s1">&#39;mut&#39;</span><span class="p">,</span>
         <span class="s1">&#39;cov&#39;</span><span class="p">,</span>
         <span class="s1">&#39;mut_freq&#39;</span><span class="p">,</span>
         <span class="s1">&#39;cleanliness&#39;</span><span class="p">,</span>
         <span class="s1">&#39;ploidy&#39;</span><span class="p">]]</span>
        <span class="n">sample_dataframes</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">df</span><span class="p">)</span>

<span class="n">mutations_dataframe</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">sample_dataframes</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [30]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">warnings</span>
<span class="n">warnings</span><span class="o">.</span><span class="n">filterwarnings</span><span class="p">(</span><span class="s2">&quot;ignore&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [31]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">f</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">plot_mutation_counts</span><span class="p">(</span><span class="n">mutations_dataframe</span><span class="o">=</span><span class="n">mutations_dataframe</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Warning: list of control samples not defined.
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/external_mutations_14_1.png" src="_images/external_mutations_14_1.png" />
</div>
</div>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="index.html">
              <img class="logo" src="_static/logo_Large.png" alt="Logo"/>
            </a></p>
<h1 class="logo"><a href="index.html">isomut2py</a></h1>



<p class="blurb">Comprehensive detection and analysis of genomic mutations from NGS sequencing data</p>






<h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="use_cases.html">General use cases</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Importing external mutations</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#Importing-VCF-files-using-pyvcf">Importing VCF files using pyvcf</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Importing-lists-of-mutations-in-arbitrary-files">Importing lists of mutations in arbitrary files</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="PE_advanced.html">Advanced ploidy estimation</a></li>
<li class="toctree-l1"><a class="reference internal" href="postprocessing.html">Further analysis, visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="code.html">API reference</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="use_cases.html" title="previous chapter">General use cases</a></li>
      <li>Next: <a href="PE_advanced.html" title="next chapter">Advanced ploidy estimation</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2019, Orsolya Pipek.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.8.0</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.11</a>
      
      |
      <a href="_sources/external_mutations.ipynb.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>