
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Getting started &#8212; isomut2py 2.0.1 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="_static/icon.ico"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="General use cases" href="use_cases.html" />
    <link rel="prev" title="Introduction" href="introduction.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #303F9F;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #D84315;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 8ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #cfcfcf;
    border-radius: 2px;
    background: #f7f7f7;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.pngmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }
</style>
<div class="section" id="Getting-started">
<h1>Getting started<a class="headerlink" href="#Getting-started" title="Permalink to this headline">¶</a></h1>
<div class="section" id="Installation">
<h2>Installation<a class="headerlink" href="#Installation" title="Permalink to this headline">¶</a></h2>
<p>IsoMut2py can be easily installed with <code class="docutils literal notranslate"><span class="pre">pip</span></code>, Python’s own package
manager:</p>
<p><code class="docutils literal notranslate"><span class="pre">pip</span> <span class="pre">install</span> <span class="pre">isomut2py</span></code></p>
<p>This also installs all python dependencies, <em>but not samtools</em>. To
install <code class="docutils literal notranslate"><span class="pre">samtools</span></code> on your computer, follow the instructions
<a class="reference external" href="http://www.htslib.org/download/">here</a>.</p>
<div class="admonition note">
<strong>Note:</strong> If you only wish to perform postprocessing and plotting steps
on a preexisting set of mutations, you can skip the installation of
<code class="docutils literal notranslate"><span class="pre">samtools</span></code>.</div>
</div>
<div class="section" id="Basic-mutation-detection-on-a-set-of-10-samples">
<h2>Basic mutation detection on a set of 10 samples<a class="headerlink" href="#Basic-mutation-detection-on-a-set-of-10-samples" title="Permalink to this headline">¶</a></h2>
<div class="section" id="Preparations,-downloading-an-example-dataset">
<h3>Preparations, downloading an example dataset<a class="headerlink" href="#Preparations,-downloading-an-example-dataset" title="Permalink to this headline">¶</a></h3>
<p>As a first step we import IsoMut2py, which also compiles the C scripts
used for mutation detection.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">isomut2py</span> <span class="k">as</span> <span class="nn">im2</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Compiling C scripts...
Done.
</pre></div></div>
</div>
<p>Next, we download an example dataset of 10 samples from the DT40 cell
line. (The BAM files in this dataset only contain reads that have been
aligned to the first chromosome of the reference genome for faster
mutation detection.)</p>
<p>The variable <code class="docutils literal notranslate"><span class="pre">exampleDataDir</span></code> stores the path to the directory where
you wish to download the example dataset. Make sure to sets its value
appropriately.</p>
<div class="admonition note">
<strong>Note:</strong> Downloading can take a while, if you have the example data
already downloaded, skip this step.</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">exampleDataDir</span> <span class="o">=</span> <span class="s1">&#39;isomut2py_download_dir/&#39;</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">im2</span><span class="o">.</span><span class="n">examples</span><span class="o">.</span><span class="n">download_example_data</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="n">exampleDataDir</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Downloading file from URL &#34;http://genomics.hu/tools/isomut2py/isomut2py_example_dataset.tar.gz&#34; to isomut2py_download_dir/isomut2py_example_dataset.tar.gz
File size: 2.1GiB
Downloading might take a while...
Download completed in 0 day(s), 11 hour(s), 38 min(s), 32 sec(s).
----------------------------------------
Extracting downloaded file to isomut2py_download_dir
Extracting completed in 0 day(s), 0 hour(s), 0 min(s), 59 sec(s).
</pre></div></div>
</div>
<p>And now we load the basic settings for the example dataset into a
directory with:</p>
<div class="admonition note">
<strong>Note:</strong> Make sure to set the value of the <code class="docutils literal notranslate"><span class="pre">exampleResultsDir</span></code> to a
path where fairly large temporary files can be stored.</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">exampleResultsDir</span><span class="o">=</span><span class="s1">&#39;isomut2py_results_dir&#39;</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">param</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">examples</span><span class="o">.</span><span class="n">load_example_mutdet_settings</span><span class="p">(</span><span class="n">example_data_path</span> <span class="o">=</span> <span class="n">exampleDataDir</span><span class="p">,</span>
                                                  <span class="n">output_dir</span> <span class="o">=</span> <span class="n">exampleResultsDir</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="Mutation-detection">
<h3>Mutation detection<a class="headerlink" href="#Mutation-detection" title="Permalink to this headline">¶</a></h3>
<p>Mutation detection can be performed with the commands below.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">mutDet</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">mutationcalling</span><span class="o">.</span><span class="n">MutationCaller</span><span class="p">(</span><span class="o">**</span><span class="n">param</span><span class="p">)</span>
<span class="n">mutDet</span><span class="o">.</span><span class="n">run_isomut2_mutdet</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
File no_ploidy_info does not exist. Using constant default ploidy of 2.
2018-12-05 10:30:57 - Mutation detection with IsoMut2


        2018-12-05 10:30:57 - Running IsoMut2 without local realignment:


                2018-12-05 10:30:57 - Preparations for parallelization:

                        2018-12-05 10:30:57 - Defining parallel blocks ...
                        2018-12-05 10:30:57 - Done

                2018-12-05 10:30:57 - (All output files will be written to isomut2py_results_dir)

                2018-12-05 10:30:57 - Generating temporary files, total number of blocks to run: 101

                2018-12-05 10:30:57 - Currently running: 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 101
                2018-12-05 10:33:00 - Finished this round.

                2018-12-05 10:33:00 - Finalizing output...
                2018-12-05 10:33:00 - Cleaning up temporary files...

2018-12-05 10:33:01 - IsoMut2 finished. (0 day(s), 0 hour(s), 2 min(s), 4 sec(s).)
</pre></div></div>
</div>
</div>
<div class="section" id="Further-optimization">
<h3>Further optimization<a class="headerlink" href="#Further-optimization" title="Permalink to this headline">¶</a></h3>
<p>The results of the mutation detection are stored in files
<code class="docutils literal notranslate"><span class="pre">all_SNVs.isomut2</span></code> and <code class="docutils literal notranslate"><span class="pre">all_indels.isomut2</span></code> in
<code class="docutils literal notranslate"><span class="pre">exampleResultsDir</span></code>. These are essentially raw, unfiltered results of
the mutation detection pipeline that can be refined with the
<code class="docutils literal notranslate"><span class="pre">optimize_results()</span></code> command. For this, one has to supply a list of
control samples where no unique mutations are expected to be found.
(These can be starting clones of a sample group from which other samples
are also available or normal samples of tumor-normal sample pairs or the
same biological sample sequenced multiple times.)</p>
<p>In our case, file pairs (<code class="docutils literal notranslate"><span class="pre">S03.bam</span></code>, <code class="docutils literal notranslate"><span class="pre">S05.bam</span></code>) and (<code class="docutils literal notranslate"><span class="pre">S08.bam</span></code>,
<code class="docutils literal notranslate"><span class="pre">S10.bam</span></code>) are two independent sequencing results of the same
biological sample, thus no unique mutations should be present in any of
these.</p>
<p>Setting these files as control samples and allowing no mutations in
these samples filters other mutations as well, based on their score
value:</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">mutDet</span><span class="o">.</span><span class="n">optimize_results</span><span class="p">(</span><span class="n">control_samples</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;S03.bam&#39;</span><span class="p">,</span> <span class="s1">&#39;S05.bam&#39;</span><span class="p">,</span> <span class="s1">&#39;S08.bam&#39;</span><span class="p">,</span> <span class="s1">&#39;S10.bam&#39;</span><span class="p">],</span> <span class="n">FPs_per_genome</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="Interpreting-the-results,-figure-generation">
<h3>Interpreting the results, figure generation<a class="headerlink" href="#Interpreting-the-results,-figure-generation" title="Permalink to this headline">¶</a></h3>
<p>Now the final set of mutations can be further investigated.</p>
<p>First it is generally a good idea to check the number of (unique)
mutations in each investigated sample. Plotting only unique mutations
allows us to check that indeed none of these were found in control
samples. This can be achieved with the following command:</p>
<div class="admonition note">
<strong>Note:</strong> Calling the magic command <code class="docutils literal notranslate"><span class="pre">%matplotlib</span> <span class="pre">inline</span></code> simply
ensures that figures are visible in the jupyter notebook environment.</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="o">%</span><span class="k">matplotlib</span> inline
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">f</span> <span class="o">=</span> <span class="n">mutDet</span><span class="o">.</span><span class="n">plot_mutation_counts</span><span class="p">(</span><span class="n">unique_only</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/getting_started_17_0.png" src="_images/getting_started_17_0.png" />
</div>
</div>
<p>Given that the investigation was confined to the first chromosome, very
meaningful conclusions should not be drawn from this example. The number
of indels is extremely low, but in the case of SNVs it is apparent that
control samples lack unique mutations, while other samples do not.</p>
<p>Further plotting options and additional tools for interpreting the
results are discussed in <a class="reference internal" href="postprocessing.html"><span class="doc">Further analysis,
visualization</span></a>.</p>
<p>An already existing set of mutation caller results can also be analysed
with isomut2py. On how to work with external files, see <a class="reference internal" href="external_mutations.html"><span class="doc">Importing
external mutations</span></a>.</p>
<p>On how to perform ploidy estimation and use the results for fine-tuning
mutation detection, see <a class="reference internal" href="PE_advanced.html"><span class="doc">Advanced ploidy
estimation</span></a>.</p>
</div>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="index.html">
              <img class="logo" src="_static/logo_Large.png" alt="Logo"/>
            </a></p>
<h1 class="logo"><a href="index.html">isomut2py</a></h1>



<p class="blurb">Comprehensive detection and analysis of genomic mutations from NGS sequencing data</p>






<h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Getting started</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#Installation">Installation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Basic-mutation-detection-on-a-set-of-10-samples">Basic mutation detection on a set of 10 samples</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="use_cases.html">General use cases</a></li>
<li class="toctree-l1"><a class="reference internal" href="external_mutations.html">Importing external mutations</a></li>
<li class="toctree-l1"><a class="reference internal" href="PE_advanced.html">Advanced ploidy estimation</a></li>
<li class="toctree-l1"><a class="reference internal" href="postprocessing.html">Further analysis, visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="code.html">API reference</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="introduction.html" title="previous chapter">Introduction</a></li>
      <li>Next: <a href="use_cases.html" title="next chapter">General use cases</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2019, Orsolya Pipek.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.8.0</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.11</a>
      
      |
      <a href="_sources/getting_started.ipynb.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>