
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Welcome to isomut2py’s documentation! &#8212; isomut2py 2.0.1 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="_static/icon.ico"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Introduction" href="introduction.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="welcome-to-isomut2py-s-documentation">
<h1>Welcome to isomut2py’s documentation!<a class="headerlink" href="#welcome-to-isomut2py-s-documentation" title="Permalink to this headline">¶</a></h1>
<p>IsoMut2py is an easy-to-use tool for the detection and postprocessing of mutations from raw NGS sequencing data. It takes sets of aligned short reads (BAM files) as its input and can explore and compare the karyotypes of different samples, detect single nucleotide variations (SNVs), insertions and deletions (indels) in single or multiple samples, optimize the identified mutations whenever provided with a list of control samples, plot mutation counts and spectra on readily interpretable charts and decompose them to predefined reference signatures.</p>
<p>IsoMut2py is an updated version of the original <a class="reference external" href="http://github.com/genomicshu/isomut/">IsoMut</a> software, mainly implemented in python. The most time-consuming parts of the workflow are however written in C, but also accessible through python wrapper functions.</p>
<p>Contents:</p>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction</a><ul>
<li class="toctree-l2"><a class="reference internal" href="introduction.html#IsoMut2py:-a-python-module-for-the-comprehensive-detection-and-analysis-of-genomic-mutations">IsoMut2py: a python module for the comprehensive detection and analysis of genomic mutations</a></li>
<li class="toctree-l2"><a class="reference internal" href="introduction.html#Features">Features</a></li>
<li class="toctree-l2"><a class="reference internal" href="introduction.html#Dependencies">Dependencies</a></li>
<li class="toctree-l2"><a class="reference internal" href="introduction.html#Authors">Authors</a></li>
<li class="toctree-l2"><a class="reference internal" href="introduction.html#How-to-cite">How to cite</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a><ul>
<li class="toctree-l2"><a class="reference internal" href="getting_started.html#Installation">Installation</a></li>
<li class="toctree-l2"><a class="reference internal" href="getting_started.html#Basic-mutation-detection-on-a-set-of-10-samples">Basic mutation detection on a set of 10 samples</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="use_cases.html">General use cases</a><ul>
<li class="toctree-l2"><a class="reference internal" href="use_cases.html#General-steps-for-analyzing-various,-differently-treated-cell-lines">General steps for analyzing various, differently treated cell lines</a></li>
<li class="toctree-l2"><a class="reference internal" href="use_cases.html#Detecting-mutations-shared-among-samples">Detecting mutations shared among samples</a></li>
<li class="toctree-l2"><a class="reference internal" href="use_cases.html#Using-local-realignment">Using local realignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="use_cases.html#Optimization-of-mutation-calls-with-control-samples">Optimization of mutation calls with control samples</a></li>
<li class="toctree-l2"><a class="reference internal" href="use_cases.html#Checking-original-sequencing-data-in-ambiguous-positions">Checking original sequencing data in ambiguous positions</a></li>
<li class="toctree-l2"><a class="reference internal" href="use_cases.html#Analysing-aneuploid-cell-lines">Analysing aneuploid cell lines</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="external_mutations.html">Importing external mutations</a><ul>
<li class="toctree-l2"><a class="reference internal" href="external_mutations.html#Importing-VCF-files-using-pyvcf">Importing VCF files using pyvcf</a></li>
<li class="toctree-l2"><a class="reference internal" href="external_mutations.html#Importing-lists-of-mutations-in-arbitrary-files">Importing lists of mutations in arbitrary files</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="PE_advanced.html">Advanced ploidy estimation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="PE_advanced.html#Ploidy-estimation-from-scratch">Ploidy estimation from scratch</a></li>
<li class="toctree-l2"><a class="reference internal" href="PE_advanced.html#Visualizing-karyotypes">Visualizing karyotypes</a></li>
<li class="toctree-l2"><a class="reference internal" href="PE_advanced.html#Comparing-samples">Comparing samples</a></li>
<li class="toctree-l2"><a class="reference internal" href="PE_advanced.html#Correcting-bad-fits,-loading-fitted-coverage-distributions">Correcting bad fits, loading fitted coverage distributions</a></li>
<li class="toctree-l2"><a class="reference internal" href="PE_advanced.html#Using-estimated-ploidies-as-input-for-mutation-detection">Using estimated ploidies as input for mutation detection</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="postprocessing.html">Further analysis, visualization</a><ul>
<li class="toctree-l2"><a class="reference internal" href="postprocessing.html#Recovering-mutations">Recovering mutations</a></li>
<li class="toctree-l2"><a class="reference internal" href="postprocessing.html#Plotting-mutations-on-rainfall-plots">Plotting mutations on rainfall plots</a></li>
<li class="toctree-l2"><a class="reference internal" href="postprocessing.html#Plotting-a-simple-hierarchical-clustering-of-the-samples">Plotting a simple hierarchical clustering of the samples</a></li>
<li class="toctree-l2"><a class="reference internal" href="postprocessing.html#Plotting-the-spectra-of-mutations">Plotting the spectra of mutations</a></li>
<li class="toctree-l2"><a class="reference internal" href="postprocessing.html#Decomposing-mutational-spectra-to-reference-signatures">Decomposing mutational spectra to reference signatures</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="code.html">API reference</a><ul>
<li class="toctree-l2"><a class="reference internal" href="code_mutation_calling.html">MutationCaller object</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_ploidy_estimation.html">PloidyEstimator object</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_plotting_functions.html">Functions for plotting</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_postprocessing_functions.html">Functions for postprocessing lists of mutations</a></li>
<li class="toctree-l2"><a class="reference internal" href="code_technical_functions.html">Technical (formatting, IO, etc.) functions</a></li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section" id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></li>
<li><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></li>
<li><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></li>
</ul>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="#">
              <img class="logo" src="_static/logo_Large.png" alt="Logo"/>
            </a></p>
<h1 class="logo"><a href="#">isomut2py</a></h1>



<p class="blurb">Comprehensive detection and analysis of genomic mutations from NGS sequencing data</p>






<h3>Navigation</h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="use_cases.html">General use cases</a></li>
<li class="toctree-l1"><a class="reference internal" href="external_mutations.html">Importing external mutations</a></li>
<li class="toctree-l1"><a class="reference internal" href="PE_advanced.html">Advanced ploidy estimation</a></li>
<li class="toctree-l1"><a class="reference internal" href="postprocessing.html">Further analysis, visualization</a></li>
<li class="toctree-l1"><a class="reference internal" href="code.html">API reference</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="#">Documentation overview</a><ul>
      <li>Next: <a href="introduction.html" title="next chapter">Introduction</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2019, Orsolya Pipek.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.8.0</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.11</a>
      
      |
      <a href="_sources/index.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>