
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Further analysis, visualization &#8212; isomut2py 2.0.1 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="_static/icon.ico"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="API reference" href="code.html" />
    <link rel="prev" title="Advanced ploidy estimation" href="PE_advanced.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #303F9F;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #D84315;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 8ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #cfcfcf;
    border-radius: 2px;
    background: #f7f7f7;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.pngmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }
</style>
<div class="section" id="Further-analysis,-visualization">
<h1>Further analysis, visualization<a class="headerlink" href="#Further-analysis,-visualization" title="Permalink to this headline">¶</a></h1>
<p>Once a basic set of mutations has been recovered for the analysed sample
set, one can proceed with additional visualization and postprocessing
steps. These are demonstrated here.</p>
<div class="section" id="Recovering-mutations">
<h2>Recovering mutations<a class="headerlink" href="#Recovering-mutations" title="Permalink to this headline">¶</a></h2>
<p>There are three options to produce a dataframe which stores all the
mutations present in the investigated sample set:</p>
<ol class="arabic simple">
<li>Run the <code class="docutils literal notranslate"><span class="pre">isomut2py</span></code> mutation detection pipeline. (See <a class="reference internal" href="getting_started.html"><span class="doc">Getting
started</span></a>.)</li>
<li>Load the list of mutations previously detected by <code class="docutils literal notranslate"><span class="pre">isomut2py</span></code>.</li>
<li>Import mutations detected by external tools. (See <a class="reference internal" href="external_mutations.html"><span class="doc">Importing external
mutations</span></a>.)</li>
</ol>
<p>For now, we will use the second option with a dataframe provided in the
example dataset. (For instructions on download, see <a class="reference internal" href="getting_started.html"><span class="doc">Getting
started</span></a>.) The reference genome used for this
set of mutations can be found in the same directory.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [73]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">isomut2py</span> <span class="k">as</span> <span class="nn">im2</span>
<span class="o">%</span><span class="k">matplotlib</span> inline
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [74]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">exampleDataDir</span><span class="o">=</span><span class="s1">&#39;/nagyvinyok/adat83/sotejedlik/orsi/&#39;</span>
</pre></div>
</div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [81]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">mutations_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">exampleDataDir</span><span class="o">+</span><span class="s1">&#39;isomut2py_example_dataset/Vizualisation/mutations_dataframe.csv&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="Plotting-mutations-on-rainfall-plots">
<h2>Plotting mutations on rainfall plots<a class="headerlink" href="#Plotting-mutations-on-rainfall-plots" title="Permalink to this headline">¶</a></h2>
<p>(To simply plot the number of mutations in each sample, see <a class="reference internal" href="getting_started.html"><span class="doc">Getting
started</span></a>.)</p>
<p>In a rainfall plot, the horizontal axis represents the genomic position
of a mutations, while the vertical axis shows the distance from the
previous mutation. Thus mutational clusters appear on the bottom part of
the rainfall plots as a cluster of dots.</p>
<p>These can be plotted with:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [86]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">f</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">plot_rainfall</span><span class="p">(</span><span class="n">mutations_dataframe</span><span class="o">=</span><span class="n">mutations_df</span><span class="p">,</span>
                           <span class="n">ref_fasta</span><span class="o">=</span><span class="n">exampleDataDir</span><span class="o">+</span><span class="s1">&#39;isomut2py_example_dataset/Vizualisation/chr1.fa&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_5_0.png" src="_images/postprocessing_5_0.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_5_1.png" src="_images/postprocessing_5_1.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_5_2.png" src="_images/postprocessing_5_2.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_5_3.png" src="_images/postprocessing_5_3.png" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_5_4.png" src="_images/postprocessing_5_4.png" />
</div>
</div>
</div>
<div class="section" id="Plotting-a-simple-hierarchical-clustering-of-the-samples">
<h2>Plotting a simple hierarchical clustering of the samples<a class="headerlink" href="#Plotting-a-simple-hierarchical-clustering-of-the-samples" title="Permalink to this headline">¶</a></h2>
<p>A naive hierarchical clustering of the analysed samples based on the
number of shared mutations can be plotted with:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [88]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">f</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">plot_hierarchical_clustering</span><span class="p">(</span><span class="n">mutations_dataframe</span><span class="o">=</span><span class="n">mutations_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_7_0.png" src="_images/postprocessing_7_0.png" />
</div>
</div>
</div>
<div class="section" id="Plotting-the-spectra-of-mutations">
<h2>Plotting the spectra of mutations<a class="headerlink" href="#Plotting-the-spectra-of-mutations" title="Permalink to this headline">¶</a></h2>
<p>The spectra of different mutation types are defined in <a class="reference external" href="https://www.biorxiv.org/content/early/2018/05/15/322859">this
paper</a>.</p>
<p>To calculate and plot the spectra of SNVs in the samples, use:</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [83]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">SNV_spectra</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">postprocess</span><span class="o">.</span><span class="n">calculate_SNV_spectrum</span><span class="p">(</span><span class="n">ref_fasta</span><span class="o">=</span><span class="n">exampleDataDir</span><span class="o">+</span><span class="s1">&#39;isomut2py_example_dataset/Vizualisation/chr1.fa&#39;</span><span class="p">,</span>
                                                     <span class="n">mutations_dataframe</span><span class="o">=</span><span class="n">mutations_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [85]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">f</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">plot_SNV_spectrum</span><span class="p">(</span><span class="n">spectrumDict</span><span class="o">=</span><span class="n">SNV_spectra</span><span class="p">,</span> <span class="n">normalize_to_1</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_10_0.png" src="_images/postprocessing_10_0.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_10_1.png" src="_images/postprocessing_10_1.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_10_2.png" src="_images/postprocessing_10_2.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_10_3.png" src="_images/postprocessing_10_3.png" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_10_4.png" src="_images/postprocessing_10_4.png" />
</div>
</div>
<p>To get the DNV spectra in a linear and a heatmap format, use:</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [89]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">DNVspectra</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">postprocess</span><span class="o">.</span><span class="n">calculate_DNV_spectrum</span><span class="p">(</span><span class="n">mutations_dataframe</span><span class="o">=</span><span class="n">mutations_df</span><span class="p">)</span>
<span class="n">DNVmatrix</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">postprocess</span><span class="o">.</span><span class="n">calculate_DNV_matrix</span><span class="p">(</span><span class="n">mutations_dataframe</span><span class="o">=</span><span class="n">mutations_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [90]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">f</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">plot_DNV_spectrum</span><span class="p">(</span><span class="n">spectrumDict</span><span class="o">=</span><span class="n">DNVspectra</span><span class="p">,</span> <span class="n">normalize_to_1</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_13_0.png" src="_images/postprocessing_13_0.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_13_1.png" src="_images/postprocessing_13_1.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_13_2.png" src="_images/postprocessing_13_2.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_13_3.png" src="_images/postprocessing_13_3.png" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_13_4.png" src="_images/postprocessing_13_4.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [91]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">f</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">plot_DNV_heatmap</span><span class="p">(</span><span class="n">matrixDict</span><span class="o">=</span><span class="n">DNVmatrix</span><span class="p">,</span> <span class="n">normalize_to_1</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_14_0.png" src="_images/postprocessing_14_0.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_14_1.png" src="_images/postprocessing_14_1.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_14_2.png" src="_images/postprocessing_14_2.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_14_3.png" src="_images/postprocessing_14_3.png" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_14_4.png" src="_images/postprocessing_14_4.png" />
</div>
</div>
<p>And finally, to get indel spectra, use:</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [92]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">IDspecra</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">postprocess</span><span class="o">.</span><span class="n">calculate_indel_spectrum</span><span class="p">(</span><span class="n">ref_fasta</span><span class="o">=</span><span class="n">exampleDataDir</span><span class="o">+</span><span class="s1">&#39;isomut2py_example_dataset/Vizualisation/chr1.fa&#39;</span><span class="p">,</span>
                                                     <span class="n">mutations_dataframe</span><span class="o">=</span><span class="n">mutations_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [93]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">f</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">plot_indel_spectrum</span><span class="p">(</span><span class="n">spectrumDict</span><span class="o">=</span><span class="n">IDspecra</span><span class="p">,</span> <span class="n">normalize_to_1</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_17_0.png" src="_images/postprocessing_17_0.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_17_1.png" src="_images/postprocessing_17_1.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_17_2.png" src="_images/postprocessing_17_2.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_17_3.png" src="_images/postprocessing_17_3.png" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_17_4.png" src="_images/postprocessing_17_4.png" />
</div>
</div>
</div>
<div class="section" id="Decomposing-mutational-spectra-to-reference-signatures">
<h2>Decomposing mutational spectra to reference signatures<a class="headerlink" href="#Decomposing-mutational-spectra-to-reference-signatures" title="Permalink to this headline">¶</a></h2>
<p>Using the consensus signatures described in the paper above, we can
decompose mutational spectra to weighted contributions of the reference
signatures. These can be achieved with:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [94]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">f</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">postprocess</span><span class="o">.</span><span class="n">decompose_SNV_spectra</span><span class="p">(</span><span class="n">SNVspectrumDict</span><span class="o">=</span><span class="n">SNV_spectra</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_19_0.png" src="_images/postprocessing_19_0.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_19_1.png" src="_images/postprocessing_19_1.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_19_2.png" src="_images/postprocessing_19_2.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_19_3.png" src="_images/postprocessing_19_3.png" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_19_4.png" src="_images/postprocessing_19_4.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [95]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">f</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">postprocess</span><span class="o">.</span><span class="n">decompose_DNV_spectra</span><span class="p">(</span><span class="n">DNVspectrumDict</span><span class="o">=</span><span class="n">DNVspectra</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_20_0.png" src="_images/postprocessing_20_0.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_20_1.png" src="_images/postprocessing_20_1.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_20_2.png" src="_images/postprocessing_20_2.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_20_3.png" src="_images/postprocessing_20_3.png" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_20_4.png" src="_images/postprocessing_20_4.png" />
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre>
<span></span>In [96]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">f</span> <span class="o">=</span> <span class="n">im2</span><span class="o">.</span><span class="n">postprocess</span><span class="o">.</span><span class="n">decompose_indel_spectra</span><span class="p">(</span><span class="n">IDspectrumDict</span><span class="o">=</span><span class="n">IDspecra</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_21_0.png" src="_images/postprocessing_21_0.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_21_1.png" src="_images/postprocessing_21_1.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_21_2.png" src="_images/postprocessing_21_2.png" />
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_21_3.png" src="_images/postprocessing_21_3.png" />
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="_images/postprocessing_21_4.png" src="_images/postprocessing_21_4.png" />
</div>
</div>
<p>The set of reference signatures used can be modified by setting the
argument <code class="docutils literal notranslate"><span class="pre">signatures_file</span></code> in the above functions. The set of
mutations included in the decomposition can be set by using the
<code class="docutils literal notranslate"><span class="pre">use_signatures</span></code> and <code class="docutils literal notranslate"><span class="pre">ignore_signatures</span></code> arguments.</p>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="index.html">
              <img class="logo" src="_static/logo_Large.png" alt="Logo"/>
            </a></p>
<h1 class="logo"><a href="index.html">isomut2py</a></h1>



<p class="blurb">Comprehensive detection and analysis of genomic mutations from NGS sequencing data</p>






<h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="use_cases.html">General use cases</a></li>
<li class="toctree-l1"><a class="reference internal" href="external_mutations.html">Importing external mutations</a></li>
<li class="toctree-l1"><a class="reference internal" href="PE_advanced.html">Advanced ploidy estimation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Further analysis, visualization</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#Recovering-mutations">Recovering mutations</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Plotting-mutations-on-rainfall-plots">Plotting mutations on rainfall plots</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Plotting-a-simple-hierarchical-clustering-of-the-samples">Plotting a simple hierarchical clustering of the samples</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Plotting-the-spectra-of-mutations">Plotting the spectra of mutations</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Decomposing-mutational-spectra-to-reference-signatures">Decomposing mutational spectra to reference signatures</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="code.html">API reference</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="PE_advanced.html" title="previous chapter">Advanced ploidy estimation</a></li>
      <li>Next: <a href="code.html" title="next chapter">API reference</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2019, Orsolya Pipek.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.8.0</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.11</a>
      
      |
      <a href="_sources/postprocessing.ipynb.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>